test/api/capi/test_capi_profiling.cpp

```diff
@@ -0,0 +1,132 @@
+#include "capi_tester.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+string BuildProfilingSettingsString(const std::vector<string> &settings) {
+	string result = "'{";
+	for (idx_t i = 0; i < settings.size(); i++) {
+		result += "\"" + settings[i] + "\": \"true\"";
+		if (i < settings.size() - 1) {
+			result += ", ";
+		}
+	}
+	result += "}'";
+	return result;
+}
+
+void RetrieveAllMetrics(duckdb_profiling_info profiling_info, const std::vector<string> &settings) {
+	for (size_t i = 0; i < settings.size(); i++) {
+		auto value = duckdb_profiling_info_get_value(profiling_info, settings[i].c_str());
+		if (value != nullptr) {
+			if (settings[i] == "EXTRA_INFO") {
+				REQUIRE(value[0] == '\"');
+				REQUIRE(value[strlen(value) - 1] == '\"');
+				duckdb_free((void *)value);
+				continue;
+			}
+			double result = 0;
+			try {
+				result = std::stod(value);
+			} catch (std::invalid_argument &e) {
+				REQUIRE(false);
+			}
+
+			duckdb_free((void *)value);
+			REQUIRE(result >= 0);
+		}
+	}
+}
+
+// Traverse the tree and retrieve all metrics
+void TraverseTree(duckdb_profiling_info profiling_info, const std::vector<string> &settings, bool is_root = true) {
+	if (is_root) {
+		// At the root, only the query name is available
```

Comment:
I remember some `JSON` outputs having quite some information in the root node. How can we access that? Or is this traversed in `RetrieveAllMetrics`?
```
{
   "name":  "Query", 
   "result": 0.490592,
   "timing": 0.490592,
   "cardinality": 0,
   "extra-info": "SELECT     c_name AS customer_name,     c_address AS customer_address,     p.p_name AS part_name,     SUM(l.l_quantity * p.p_retailprice) AS total_spent FROM     customer c         JOIN     orders o ON c.c_custkey = o.o_custkey         JOIN     lineitem l ON o.o_orderkey = l.l_orderkey         JOIN     part p ON l.l_partkey = p.p_partkey WHERE     c_nationkey = '1'   AND o_orderdate BETWEEN '1990-01-01' AND '1995-12-31' GROUP BY     c.c_custkey, c_name, c_address, part_name HAVING     total_spent > 100000 ORDER BY     total_spent DESC;", 
   "timings": [
   {
   "annotation": "optimizer", 
   "timing": 0.001389
   },
   {
   "annotation": "optimizer > column_lifetime", 
   "timing": 0.000037
   },
   {
...
```

Reply:
Yes, that sounds good. Let's add it as a field. From my understanding, it is not a setting. So, because of that it would have to be a field? And a small nit: Can we lowercase `query` to be consistent with the other fields?